dags/springer/repository.py

```diff
@@ -44,8 +54,15 @@ def __find_all_extracted_files(self):
         return [
             f.key
             for f in self.s3.objects.filter(Prefix=self.EXTRACTED_DIR).all()
-            if self.__file_is_meta(f.key) or ".pdf" in f.key
+            if self.is_meta(f.key) or ".pdf" in f.key
+        ]
+
+    def __find_given_filenames(self, filenames):
+        return [
+            f.key
+            for f in self.s3.objects.filter(Prefix=self.EXTRACTED_DIR).all()
+            if f.key.split("/")[-1] in filenames
```

Comment:
why not try to get the object directly instead of listing everything?

Reply:
Done